metagraph/src/annotation/int_matrix/extended/csc_matrix.hpp

```diff
@@ -57,62 +57,62 @@ class ColumnRankMatrix : public CountMatrix {
 
   private:
     BaseMatrix binary_matrix_;
-    std::vector<sdsl::int_vector<>> count_columns_;
+    std::vector<sdsl::int_vector<>> column_values_;
 };
 
 
 template <class BaseMatrix>
-inline typename ColumnRankMatrix<BaseMatrix>::ColumnCounts
-ColumnRankMatrix<BaseMatrix>::get_row_counts(Row row) const {
-    const auto &column_ranks = binary_matrix_.get_row_ranks(row);
-    ColumnCounts row_counts;
-    row_counts.reserve(column_ranks.size());
+inline typename CSCMatrix<BaseMatrix>::RowValues
+CSCMatrix<BaseMatrix>::get_row_values(Row row) const {
+    const auto &column_ranks = binary_matrix_.get_column_ranks(row);
```

Comment:
`binary_matrix_.get_column_ranks` returns an r-value, won't this cause issues?
```suggestion
    auto column_ranks = binary_matrix_.get_column_ranks(row);
```

Reply:
Assigning const references to returned objects is always correct.
https://stackoverflow.com/questions/7340925/return-by-value-assigned-to-const-reference/7340943

I think it's a bit better because this implies that the object is not changed in this function, hence the function can, in principle, return const references as well, and no copy would be made in that case.